### PR TITLE
Updated exponential model

### PR DESCRIPTION
**Pull request by mitchell-thomann**
Updated exponential model - including:

- Updated parameterization to match DoseFinding package
- Included upper bound on truncated prior to be user specified in case of sampling errors
- Updated documentation with new parameterization and truncated prior

### CODE REVIEW

File: R/model.R [hunk 1]
```diff
@@ -523,12 +527,14 @@ model_emax_binary <- function(
 }
 
 #' @rdname model
+#' @param bound is the upper truncation bound of the exponential scale parameter \eqn{b_3}
 #' @section Exponential Binary:
-#'   \deqn{y \sim Binomial(n, f(d))}
-#'   \deqn{link(f(d)) = b_1 + b_2 * (exp(b_3 * d) - 1)}
+#'   \deqn{y \sim N(f(d), \sigma^2)}
```
Comment (rich-payne) [inline R/model.R:532]: This is a binary function, so needs to retain the binomial likelihood and description.

**Review by rich-payne — CHANGES_REQUESTED** [review]
Right now the PR is incomplete for the pr_med() etc. functions.  However, if we plan to deprecate, this won't matter.

File: R/model.R [hunk 2]
```diff
@@ -233,6 +235,7 @@ model_exp <- function(
   sigma_b3,
   shape,
   rate,
+  bound = 10,
```
Comment (rich-payne) [inline R/model.R:238]: How universally applicable will this bound be?  We don't specify any other hyperparameters.  For consistency, should we omit a default argument?

File: inst/jags/likelihood_emax_binary.jags [hunk 3]
```diff
@@ -1,4 +1,5 @@
 for(i in 1:n_obs) {
-  LINK(f[i]) <- (ifelse(dose[i] == 0, a + b1 * (LONGITUDINAL), a + (b1 + (b2 - b1) / (exp(b4 * (b3 - log(dose[i]))) + 1)) * (LONGITUDINAL))) 
+ # LINK(f[i]) <- (ifelse(dose[i] == 0, a + b1 * (LONGITUDINAL), a + (b1 + (b2 - b1) / (exp # (b4 * (b3 - log(dose[i]))) + 1)) * (LONGITUDINAL)))
```
Comment (rich-payne) [inline inst/jags/likelihood_emax_binary.jags:2]: Remove commented code

File: inst/jags/likelihood_exp.jags [hunk 4]
```diff
@@ -1,6 +1,7 @@
 # likelihood for sufficient means
 for(i in 1:n_sufficient_means) {
-  f_sufficient[i] <- a + (b1 + b2 * (1 - exp(- b3 * dose_sufficient[i]))) * (LONGITUDINAL)
+  #f_sufficient[i] <- a + (b1 + b2 * (1 - exp(- b3 * dose_sufficient[i]))) * (LONGITUDINAL)
```
Comment (rich-payne) [inline inst/jags/likelihood_exp.jags:3]: Remove commented code.

**Review by rich-payne — CHANGES_REQUESTED** [review]
Looking good overall.  Thank you!

A few things that I've noted in comments which need addressing.  Once addressed, please open a new PR to request a merge into the "version4" branch rather than the main branch.    We can then place all the major updates on that branch.

Please record the changes which are made in this PR into the News file.

Lastly, I think we'll need to include some warnings/deprecations on the exponential and emax models about the changes.  There are some nice functions which the tidyverse supports it.  I will see if I can find a link to them and open a new issue.

**Comment by rich-payne** [comment]
We can use a call to `rlang::warn()` and `rlang::inform()` with the `.frequency` argument to display messages once per session or once every 8 hours to let users know about the update to the models.  This should probably be within the "model" functions directly.

**Comment by mitchell-thomann** [comment]
Updated in new pull request

**Pull request closed without merging**